src/pattern_matching/myers/builder.rs

```diff
@@ -0,0 +1,187 @@
+///     (b'B', &b"CGTSYK"[..]),
+///     (b'N', &b"ACGTMRWSYKVHDB"[..])
+/// ];
+///
+/// let mut builder = MyersBuilder::new();
+///
+/// for &(base, equivalents) in &ambigs {
+///     builder.ambig(base, equivalents);
+/// }
+///
+/// let text = b"GGATGNGCGCCATAG";
+/// let pattern = b"TRANCGG";
+/// //                *   * (mismatch)
+///
+/// let myers = builder.build(pattern);
+/// assert_eq!(myers.distance(text), 2);
+/// # }
+/// ```
+///
+/// Note that only ambiguities in the pattern are recognized. The reverse is not true; ambiguities
+/// in the search text are not matched by multiple symbols in the pattern. This would require
+/// specifying additional ambiguities (`builder.ambig(b'A', b"MRWVHDN")`, etc...).
+#[derive(Default, Clone, Eq, PartialEq)]
+pub struct MyersBuilder {
+    ambigs: HashMap<u8, Vec<u8>>,
+    wildcards: Vec<u8>,
+}
+
+impl MyersBuilder {
+    pub fn new() -> MyersBuilder {
+        Self::default()
+    }
+
+    /// Allows to specify ambiguous symbols and their equivalents. Note that the ambiguous symbol
+    /// will always be matched by itself. Explicitly including it in the equivalents is not
+    /// necessary.
+    ///
+    /// # Example:
+    ///
+    /// ```
+    /// # extern crate bio;
+    /// use bio::pattern_matching::myers::MyersBuilder;
+    ///
+    /// # fn main() {
+    /// let text = b"ACCGTGGATGAGCGCCATAG";
+    /// let pattern =      b"TGAGCGN";
+    ///
+    /// let myers = MyersBuilder::new()
+    ///     .ambig(b'N', b"ACGT")
+    ///     .build(pattern);
+    ///
+    /// assert_eq!(myers.distance(text), 0);
+    /// # }
+    pub fn ambig<I, B>(&mut self, byte: u8, equivalents: I) -> &mut Self
+    where
+        I: IntoIterator<Item = B>,
+        B: Borrow<u8>,
+    {
+        let eq = equivalents
+            .into_iter()
+            .map(|b| *b.borrow())
+            .chain(Some(byte))
+            .collect();
+        self.ambigs.insert(byte, eq);
+        self
+    }
+
+    /// Allows to specify a wildcard symbol, that upon appearance in the search text
+    /// shall be matched by any symbol of the pattern. Multiple wildcards are possible.
+    /// For the inverse, that is, wildcards in the pattern matching any symbol in search
+    /// text, use `ambig(byte, 0..255)`.
+    ///
+    /// # Example:
+    ///
+    /// ```
+    /// # extern crate bio;
+    /// use bio::pattern_matching::myers::MyersBuilder;
+    ///
+    /// # fn main() {
+    /// let text = b"ACCGTGGATGAGCG*CATAG";
+    /// let pattern =      b"TGAGCGT";
+    ///
+    /// let myers = MyersBuilder::new()
+    ///     .text_wildcard(b'*')
+    ///     .build(pattern);
+    ///
+    /// assert_eq!(myers.distance(text), 0);
+    /// # }
+    pub fn text_wildcard(&mut self, wildcard: u8) -> &mut Self {
+        self.wildcards.push(wildcard);
+        self
+    }
+
+    /// Creates a Myers instance given a pattern, using `u64` as bit vector type
+    pub fn build<'a, P>(&self, pattern: P) -> Myers<u64>
```

Comment:
I would rename build_other into this, and have two shorthands `build_64` and `build_128`.